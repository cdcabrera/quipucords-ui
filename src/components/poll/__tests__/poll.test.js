import React from 'react';
import { shallow } from 'enzyme';
import { Poll, PollCache } from '../poll';

describe('Poll Component', () => {
  it('should render a component that polls', done => {
    const props = {
      itemId: 'lorem',
      itemIdCheck: true,
      onPoll: jest.fn(),
      interval: 1
    };

    const component = shallow(<Poll {...props}>lorem ipsum</Poll>);
    expect(component).toMatchSnapshot('basic');
    expect(PollCache).toMatchSnapshot('pre cache');

    setTimeout(() => {
      component.setProps({
        itemIdCheck: false
      });

      expect(props.onPoll).toHaveBeenCalled();
      expect(PollCache).toMatchSnapshot('post cache');
      done();
    }, 3);
  });
});
