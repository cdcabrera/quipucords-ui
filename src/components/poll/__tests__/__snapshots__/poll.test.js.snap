// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll Component should render a component that polls: basic 1`] = `
<Fragment>
  lorem ipsum
</Fragment>
`;

exports[`Poll Component should render a component that polls: post cache 1`] = `
Object {
  "check": Object {
    "lorem": false,
  },
  "timer": 7,
}
`;

exports[`Poll Component should render a component that polls: pre cache 1`] = `
Object {
  "check": Object {
    "lorem": true,
  },
  "timer": 5,
}
`;
